### ISSUE TITLE
Jet Set Radio Future - broken graphics on various locations

### ISSUE THREAD

**Issue by furrya** [Cxbx-Reloaded/Cxbx-Reloaded#251]
I just have launched the game on windows 7 x 64 , and I can see normal graphic on some locations , but  on others it's broken
![screenshot_7](https://cloud.githubusercontent.com/assets/25540784/24128013/d67b2838-0e24-11e7-92c9-6de9af49c550.png)
![screenshot_9](https://cloud.githubusercontent.com/assets/25540784/24128014/d67f0cbe-0e24-11e7-959f-211dfba0f757.png)
![screenshot_11](https://cloud.githubusercontent.com/assets/25540784/24128015/d681b87e-0e24-11e7-8e8f-0ed07449c0b9.png)

**Comment by BenNottelling** [comment]
Looking like windows 7 is the best OS for this emulator

**Comment by furrya** [comment]
With the latest commit, the graphics now is same as on Windows 10 . Wow
![screenshot_4](https://cloud.githubusercontent.com/assets/25540784/24176730/25a78e62-0ee1-11e7-8695-2b8895a4bc56.png)
![screenshot_5](https://cloud.githubusercontent.com/assets/25540784/24176731/25a8708e-0ee1-11e7-986f-e2bce1962733.png)
![screenshot_7](https://cloud.githubusercontent.com/assets/25540784/24176732/25a96c1e-0ee1-11e7-9934-c5b480f2652a.png)


on Windows 7

**Comment by furrya** [comment]
With the latest commits - Win 10 x64 
![screenshot_1](https://cloud.githubusercontent.com/assets/25540784/24318136/dfe13792-1143-11e7-8fb1-020f2c8f3d7d.png)

**Comment by bhodg22** [comment]
Does the rest of the level stay normal for you? I find it sometimes flickers to shades of black at times, also what GPU are you using

**Comment by furrya** [comment]
@bhodg22 Same behaviour on my Palit Geforce 610m, Regardless of the OS ( Win 7 x64, Win 10 x64) . Flickers appears many times on the start and ingame also

**Comment by bhodg22** [comment]
With the latest commit, I managed to get past the first cut scene and back into the garage, and the game was actually running full speed but then crashed when I tried going to the next level

**Comment by PatrickvL** [comment]
@bhodg22 : When reporting something like this, please mention exactly which commit you're referring to, by copy-pasting the URL for it, like this : https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/commit/2a8cc3722ba1eea8fe61c035149cd923f9ffe0a0 (this IS the one you're referring to, isn't it?)

**Comment by PatrickvL** [comment]
@furrya This flickering is also present in other software, specifically the Dashboard, JSRF and other titles - see https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/issues/91#issuecomment-280917520

**Comment by bhodg22** [comment]
yeah thats the one i was referring to, I didnt bother posting any details because I tried to recreate it a second time but it didnt quite make it to the end of the cutscene, might try a third time soon

**Comment by LukeUsher** [comment]
I've been able to play up to the point of the tutorial where you do the 'boost dash'. I usually get a crash at that point, but it seems to be caused by my graphics drivers.

**Comment by ObiKKa** [comment]
Urgent disaster! Needs remedying! After PR #375 (Overlay conversion) & #376 (Integrated copy OOVPAs. Most likely problem.) game crashes after company intros shown.

![2017-apr-19 _crash-after-intros](https://cloud.githubusercontent.com/assets/18428108/25144712/2feab868-24b2-11e7-8e94-3c1b659f0b41.png)
[KrnlDebug_(2017-04-19_Crash-after-intros).txt](https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/files/930565/KrnlDebug_.2017-04-19_Crash-after-intros.txt)

**Comment by PatrickvL** [comment]
If the function locations differ, that's probably the main cause of regressing. Video issues shouldn't give such hard crashes

**Comment by bhodg22** [comment]
Strange, my JSRF runs fine on that same commit

**Comment by PatrickvL** [comment]
Then maybe you're running another version of the same game

**Comment by bhodg22** [comment]
Might be similar to the futurama issues, where other people can boot the PAL version as I still cannot

**Comment by jarupxx** [comment]
@ObiKKa compare PR #374 log & #376, we will find the cause of the regression

**Comment by LukeUsher** [comment]
JSRF also still works for me. How many times did you try? JSRF only boots successfully 8/10 times for me, the other times I get a similar error you see above.

**Comment by ObiKKa** [comment]
@LukeUsher & @jarupxx, that's strange. today I restarted the merged master of PR 374 (b525828e6959bd2f57b260e784b77391cb8c3024) coz I already downloaded it before. Still worked. Then I booted up the merged master of PR 376 (77933ad8ebf261634cfc2bde978f4fb8cc14d689), the old download - could be before you made that 2nd commit after I read the page (	OOVPA integration D3D 5849 to 3925).
But never mind that, in that build I ran JSRF (US build) again and it got to the menu and in-game again.

To Luke, I only ran JSRF twice on that last build (PR 376) a day before as I was testing other games.
Seems strange that some games do fail to run forward to the same progress line for inexplicable reasons sometimes. But, yeah, all seems to be fine. I'll test the newest build again, although that's unlikely to change my experience in Innocent Tears, etc.

**Comment by ObiKKa** [comment]
Oh, after about 4 crashes (That's a lot. Luke says his own test crashes on avg 2x every 10 tries.) I was able to see the main menu screen. But lots of blackness in the background and occasionally switching to yellow/white visuals.
Then in-game, first cutscene looks similar like before. But in-game was horrible. It appears that PR #384 (Implement HLE cache) may have mucked up the visuals for me. I think it may remember the first or so crash and then made the visuals worse on later boots. It's been much black over everything in the world for half of the viewing angles. Also when it's finally back to yellow and white polygons (corrupt vertex shaders ATM) I saw one gray shader on a platform was flickering.

![collage-2017-04-21](https://cloud.githubusercontent.com/assets/18428108/25256012/37bf8eea-2671-11e7-90bb-4fda6c6339dd.png)

**Comment by PatrickvL** [comment]
I suspect the graphics are damaged because the HLE cache might be skipping the detection of render state and texture state addresses

**Comment by LukeUsher** [comment]
Sorry about that, i overlooked that part of HLE patching. Should be an easy fix, I'll hopefully get that done before Monday.

**Comment by furrya** [comment]
For me the state of the emulation gets better than in previous builds
The game looks better 
For me the state of the emulation gets better than in previous builds
The game looks better , it's looks like c1e4d973 now 

![screenshot_1](https://cloud.githubusercontent.com/assets/25540784/25300693/3869aab2-2750-11e7-960b-7cf753b87053.png)
![screenshot_2](https://cloud.githubusercontent.com/assets/25540784/25300696/3c825a9a-2750-11e7-968c-e1669ad9c7c8.png)
 1 st screen 18 Apr Build   and second Cxbx-Reloaded Version 1a9b20ad (Apr 21 2017)

**Comment by ObiKKa** [comment]
The main character model in the intro game cutscene seems to be fully rendered a bit better after PR #392 (Removed CreatePalette) and #393 (Temporarily disabled vertex buffer cache). And I could see some vehicles in the background in the main menu. But yeah loading the game takes a way long time and game runs slower as a result.
This game is easier said than done to fix up...

![collage-2017-04-24](https://cloud.githubusercontent.com/assets/18428108/25318169/7ee1f882-28cc-11e7-9695-0834ad36fdd1.png)

**Comment by bhodg22** [comment]
I also have the very slow loading and also the main menu is very slow, graphics on characters has improved alot and ingame is a bit slower, however my speedup trick still works to get the game running fullspeed

**Comment by PatrickvL** [comment]
This should now render a lot better!

**Comment by ObiKKa** [comment]
No. It has crashed many times for me. I also re-cleared the HLE cache a few times later on as well.
Now the emulation seems to be so accurate that anything a game requires is missing then it becomes so brutal to said emulation!

![2017-apr-30 _index-buffer_crash_01](https://cloud.githubusercontent.com/assets/18428108/25559393/49c8fa42-2d7c-11e7-924d-8ea6b2133589.png)
[KrnlDebug_(2017-04-30_Index-Buffer_Crash).txt](https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/files/966438/KrnlDebug_.2017-04-30_Index-Buffer_Crash.txt)

**Comment by PatrickvL** [comment]
You're getting the same results as me, while Luke is getting increasingly better results. I suspect it's related to the version we're running (NTSC/PAL , older/newer builds of the game, or whatever).

**Comment by blueshogun96** [comment]
This is the exact same crash that I would get with the old Cxbx.  Something has definitely regressed.  It's the same with both PAL and NTSC, so it's not just him.

Several other titles (most notably SEGA titles) are regressing right now, and I'll make a note of each one.

Shogun.

**Comment by LukeUsher** [comment]
Interestingly JSRF still works on my machine, in fact it's what I perform most of my testing with: I know how prone to regressions it is. Perhaps there's something different between my master and this one.

**Comment by blueshogun96** [comment]
Sadly, I'd have to assume that also.  It also reminds me of how Outrun2 appears to work for everyone else... except me!  I'll be sure to keep you updated.

Shogun.

**Comment by LukeUsher** [comment]
Should be fixed now, my new implementation of AvQueryAvCapabilities was the problem.

**Comment by PatrickvL** [comment]
That was quick Luke - thanks!

**Comment by PatrickvL** [comment]
@blueshogun96 Have you considered submitting pull requests? Bug reports are nice and all, but better code, yeah, shiver me timbers, that's what we really want! ;)

**Comment by blueshogun96** [comment]
@PatrickvL I have, but unfortunately I haven't had my own fix for such things.  Some of these errors are beyond me, especially for me being out of the loop so long. :(

Shogun.

**Comment by CakeLancelot** [comment]
In build 394aee6b the game is much more stable and I'm able to make it through most of the first cutscene, though the graphics are about the same compared to the last few builds. Seems a bit faster too though that may just be me. Excellent progress guys!
![cxbx_2017-04-30_02-20-22](https://cloud.githubusercontent.com/assets/5341136/25562492/6f388b42-2d4c-11e7-8a5e-6334912c2afa.png)
![cxbx_2017-04-30_02-23-05](https://cloud.githubusercontent.com/assets/5341136/25562491/6f376564-2d4c-11e7-9fa1-c47ff32cdc74.png)

**Comment by LukeUsher** [comment]
In my recent testing, I managed to get past the cutscene and into gameplay properly, I played up until meeting Beat. One thing that was a little strange: after the intro finished, the music and sound became perfect.

**Comment by CakeLancelot** [comment]
> In my recent testing, I managed to get past the cutscene and into gameplay properly, I played up until meeting Beat. One thing that was a little strange: after the intro finished, the music and sound became perfect.

I haven't played past dogenzaka hill, but during the time on that level the audio was still broken for me. 

On top of that as mentioned before that some of the particles and effects start working after using graffiti however this will revert when you enter the next 'scene' (includes the jumpcuts in cutscenes while loading in assets or whatever it's doing). 

Also encountered this weirdness when spraypainting a wall:
![cxbx_2017-04-30_02-37-32](https://cloud.githubusercontent.com/assets/5341136/25562839/aaceddd8-2d55-11e7-9734-56738b7796ad.png)
Then again as #271 shows there are differences in accuracy depending on graphics hardware, perhaps drivers even.

Let me know if you need any other info, my specifications are listed in #271.

**Comment by ObiKKa** [comment]
Unfortunately I'm not able to continue in the Start menu for a while after all these little and big PRs. One same error message shows up before loading those intro splash screens (Press _Ignore_ to resume) and repeatedly after a brief period of time in the Start menu (I cannot see the buttons at this time).
I think it might be worth it to tackle that 'XInputPoll' problem.

Also congratulations to @blueshogun96 for his work on getting those Panzer Dragoon game(s) up and running! :) Hopefully a fourth regular programmer on this team? Dream on!

![2017-may-03 _error-box_01](https://cloud.githubusercontent.com/assets/18428108/25628514/d0e45c36-2fa9-11e7-9a1c-2e29e995ba8f.png)

**Comment by PatrickvL** [comment]
I've seen almost the same error for a while, until I fixed a bug in D3DSurface_LockRect; it didn't account for textures, instead of surfaces being passed in

**Comment by ObiKKa** [comment]
Yes, JSRF did get back into the game after Luke's PR #424 (Remove unnecessary breakpoint. Reason: Resources are now separated from host resources so those errors kept popping up early.). But only after I tested 2 other games.
However I now see those stretched coloured graphical glitches linking the boots of my skater on a level surface as well as between the outer of either boot/leg to their adjacent when jumping high in the air.

Now, to the other two games - Namco Museum and Battlestar Galactica: No, those new error messages still pop up at the same time so they are not solved yet.

Created with photocollage.com:
![collage-2017-05-03](https://cloud.githubusercontent.com/assets/18428108/25638525/0a4ffd64-2fcc-11e7-841b-22f5321a136f.png)

**Comment by PatrickvL** [comment]
Regarding the glitches linking some elements together: this might be caused by the same issue that seems to have broken Turok (right after starting a game), probably an index buffer change. @LukeUsher Could it be that our indeed buffers are not correctly aligned?

**Comment by furrya** [comment]
Hmmm. For me same error has appears as Xinput error with Cxbx-Reloaded Version 3e872214 (May  2 2017)
![screenshot_1](https://cloud.githubusercontent.com/assets/25540784/25644250/8d814ec8-2fdf-11e7-9463-dc861658e075.png)
[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/files/972227/KrnlDebug.txt)

**Comment by bhodg22** [comment]
Was meant to post this earlier, I managed to get to the beat mission in a build from about 2 months ago and the sound was also became perfect back then as well. As for one of the latest builds (Maybe from about 10 days ago?) Alot of the ingame elements have become invisible for me (buildings, the ground, rails, etc) making it impossible for me to progress further in the games as i cant see the levels!

**Comment by CakeLancelot** [comment]
As of 2d720a1 the shading on the characters has improved, though many previous graphical issues are still present. 
![cxbx_2017-05-13_01-59-10](https://cloud.githubusercontent.com/assets/5341136/26023394/1f7c558c-3780-11e7-923e-a8b5fcaadbef.png)
vs
![cxbx_2017-04-30_02-20-22](https://cloud.githubusercontent.com/assets/5341136/26023397/2ae5caa2-3780-11e7-9a0e-d3049b89dd23.png)

**Comment by PatrickvL** [comment]
Which one is the better one? Do you have some reference material perhaps?

**Comment by CakeLancelot** [comment]
> Which one is the better one? Do you have some reference material perhaps?
Sure, here's some comparisons:
![comparison](https://cloud.githubusercontent.com/assets/5341136/26023749/d8eb8148-3788-11e7-8323-5b1840147d8a.png)
![comparison2](https://cloud.githubusercontent.com/assets/5341136/26023752/ddf6571c-3788-11e7-8154-a63d873d2559.png)

**Comment by HyperScorpio** [comment]
Funny how we spray in the cops back and it doesn´t bug the map at all. The save game graffiti appears a annoying white X in the middle and it hide you from the screen,

**Comment by PatrickvL** [comment]
That's a notification for sound related issues - @radwolfie put it in to get feedback on his work

**Comment by HyperScorpio** [comment]
Already checked today.

**Comment by PatrickvL** [comment]
What do you mean by "already checked today"? It's unclear, even in context.

**Comment by HyperScorpio** [comment]
I check @RadWolfie feedback.

**Comment by HyperScorpio** [comment]
I remove it and i create a new issue. Sorry and i know this thead is closed.

**Comment by HyperScorpio** [comment]
There is a better tool to extract xbox games. Xbox Backup Creator 2.9.0.425
Open and browse the image and extract. Works amazing with Redump ISOS.
Enjoy.

**Comment by PatrickvL** [comment]
Pleased reword the above comment. Right now it almost sounds like an advertisement to use illegally obtained dumps. We do not tolerate nor advocate piracy.

**Comment by ladsoftware** [comment]
Where may i ask luke coding questions?

**Comment by HyperScorpio** [comment]
@PatrickvL Done. And so sorry i just trying to help others.

**Comment by ladsoftware** [comment]
@HyperScorpio
Your post is very helpful. Thank you, for the post.
Ive been digging around for away to extract my jsrf dvd, without modded xbox.

**Comment by PatrickvL** [comment]
The Cxbx-Reloaded github front page links to our gitter.im chat lobby

**Comment by HyperScorpio** [comment]
@ladsoftware Just Remember.
Redumps: Xbox Backup Creator
No Redumps C-Box Tool

**Comment by ladsoftware** [comment]
@HyperScorpio
Thank you.

**Comment by Syampuuh** [comment]
Haven't seen this mentioned anywhere yet, but spraying graffiti changes all the shadows into a full white shape that you can't see out of, and only for the currently ghosted area of the map.
https://streamable.com/l9gre
Playing on the latest release build https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/commit/bbb497e2a48f7d6390a213fdb59558e45d561b02
(I'm playing with the 4k xbe edit, but it happens without it too)

**Comment by CakeLancelot** [comment]
I've copied important comments from this issue to its new place at Cxbx-Reloaded/game-compatibility#178, this one can be closed.

**Comment by C3C6** [comment]
I figured out how to solve the screen flickering issue (white). Many of you may be like me where you chose to change your video settings (under Eeprom configuration) to widescreen. DO NOT DO THIS. It will result in this issue. Instead, leave the video settings at normal. If I were you, I'd personally leave all the Eeprom stuff at their default settings. This seems to have immediately fixed the issue for me.

**Comment by C3C6** [comment]
> I just have launched the game on windows 7 x 64 , and I can see normal graphic on some locations , but on others it's broken ![screenshot_7](https://cloud.githubusercontent.com/assets/25540784/24128013/d67b2838-0e24-11e7-92c9-6de9af49c550.png) ![screenshot_9](https://cloud.githubusercontent.com/assets/25540784/24128014/d67f0cbe-0e24-11e7-959f-211dfba0f757.png) ![screenshot_11](https://cloud.githubusercontent.com/assets/25540784/24128015/d681b87e-0e24-11e7-8e8f-0ed07449c0b9.png)

I figured out how to solve the screen flickering issue (white). Many of you may be like me where you chose to change your video settings (under Eeprom configuration) to widescreen. DO NOT DO THIS. It will result in this issue. Instead, leave the video settings at normal. If I were you, I'd personally leave all the Eeprom stuff at their default settings. This seems to have immediately fixed the issue for me.
